cachi2/core/package_managers/pip.py

```diff
@@ -1299,11 +1300,15 @@ def from_line(cls, line: str, options: list[str]) -> Optional["PipRequirement"]:
         qualifiers = {}
         requirement = cls()
 
-        direct_access_kind, is_direct_access = cls._assess_direct_access_requirement(line)
+        direct_access_kind, is_direct_access = cls._assess_direct_access_requirement(
+            line, cls.URL_SCHEMES, cls.VCS_SCHEMES
```

Comment:
Looking at the code in more detail there's actually no reason for us to keep ^these as class attributes given they're only ever used in the static methods, so there's nothing preventing us from moving the class attributes to the respective static methods and define them as constants there. Same for `HAS_NAME_IN_DIRECT_ACCESS_REQUIREMENT` below.
Note that such an attribute move should be in a separate commit preceding this one converting the methods from class methods to static methods.

Reply:
Moving these inside the methods sounds great, but can you specify what you mean by "define them as constants"? Just so we're on the same page.

Reply:
In Python constants  == upper case variables, that's all there's to it :) .

Reply:
@eskultety; Is there an advantage to (currently) only compiling HAS_NAME_IN_DIRECT_ACCESS_REQUIREMENT once (when the the module containing the class is loaded)?

IOW, won't we otherwise be recompiling it on every access of `_assess_direct_access_requirement`?

(yeah, it's a micro-optimization...)

Reply:
> IOW, won't we otherwise be recompiling it on every access of _assess_direct_access_requirement?

@ben-alkov @ianrochapg Oh, missed that one, in that particular case it makes sense to keep passing it to the method as a `pattern` argument.